### DISCUSSION TITLE
Recursive State object

### DISCUSSION THREAD

**Discussion by Josef177** [pmndrs/zustand#1088] [Q&A]
Imagine that there is one large object in the Zustand store that can be infinitely immersed. The structure of this immersion is repeated and React components are rendered according to this structure of this object.

Since the depth of this recursion is dynamic, it cannot be defined directly in the code, but is created during the program's runtime.
Each React component receives part of the data in Props according to the immersion level. These components actually replicate this large state object in the UI.

So my question is: **How to achieve that this object is kept in the Zustand store and at the same time, when changing the children of some nested levels, the upper levels are not rendered unnecessarily?**

If I use, for example, Immer in this way, the object will be rerendered from top to bottom up to the immersion level, right? 
And I think the same thing happens if I use e.g. this approach:
`  immerInc: () =>
    set(produce((state: State) => { ++state.deep.nested.obj.count })),`

Thanks for any answer. :)

**Comment by dai-shi** [comment]
If I understand correctly, this would be a bit hard task for zustand. There should be some practices/workarounds though.
I would personally use [valtio](https://github.com/pmndrs/valtio) for such use cases.

Anyway, I hope someone else jump in this discussion.
